spotbugs/src/main/java/edu/umd/cs/findbugs/SourceLineAnnotation.java

```diff
@@ -944,6 +945,43 @@ public String getSourcePath() {
         return sourcePath;
     }
 
+    /**
+     * Returns the complete path of the source file the analyzed class has been generated from,
+     * if this information is available to the current AnalysisContext, otherwise falls back to getSourcePath().
+     *
+     * @return the complete path of the source file the analyzed class has been generated from,
+     * otherwise falls back to getSourcePath().
+     */
+    public String getRealSourcePath() {
+        if (isSourceFileKnown()) {
+            SourceFinder sourceFinder = getSourceFinder();
+            if (sourceFinder != null)
+            {
+                try {
+                    return new File(sourceFinder.findSourceFile(this).getFullFileName()).getCanonicalPath();
+                } catch (IOException e) {
+                    AnalysisContext.logError("Error resolving Real SourcePath (only relative source path will be available) ", e);
+                }
+            }
+            else {
+                AnalysisContext.logError("No SourceFinder found (only relative source path will be available) ");
+            }
+        }
+        return getSourcePath();
+    }
+
+    private @CheckReturnValue SourceFinder getSourceFinder() {
+        Project project = myProject.get();
+        if (project != null) {
+            return project.getSourceFinder();
+        }
+        AnalysisContext currentAnalysisContext = AnalysisContext.currentAnalysisContext();
```

Comment:
I would probably write this as

    return Optional.ofNullable(AnalysisContext.currentAnalysisContext())
        .map(AnalysisContext::getSourceFinder)
        .orElse(null);

Reply:
I  Like that one, added to code.